### ISSUE TITLE
Canvas crashes bun

### ISSUE THREAD

**Issue by 0PandaDEV** [oven-sh/bun#13171]
### How can we reproduce the crash?

"canvas": "^2.11.2"

```ts
// generate.ts

import { createCanvas } from 'canvas';
import { fetchDataForAllYears } from './github';
import fs from 'fs';

async function generateContributionImage(username) {
  const data = await fetchDataForAllYears(username, 'flat');
  const contributions = data.contributions;
  const canvasWidth = 53 * 12;
  const canvasHeight = 7 * 12;
  const canvas = createCanvas(canvasWidth, canvasHeight);
  const ctx = canvas.getContext('2d');

  ctx.fillStyle = '#ffffff';
  ctx.fillRect(0, 0, canvasWidth, canvasHeight);

  contributions.forEach(contribution => {
    const date = new Date(contribution.date);
    const x = ((date.getDay() + 1) % 7) * 12; 
    const y = Math.floor(date.getTime() / (1000 * 60 * 60 * 24 * 7)) % 53 * 12; 
    ctx.fillStyle = contribution.color;
    ctx.fillRect(x, y, 10, 10); 
  });

  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync('./output.png', buffer);
}

generateContributionImage('0pandadev').catch(console.error);
```

```zsh
bun run generate.ts
```

### Relevant log output

```shell
1 | 'use strict'
2 |
3 | const bindings = require('../build/Release/canvas.node')
          ^
TypeError: LoadLibrary failed: A dynamic link library (DLL) in
      at C:\Users\pandadev\Documents\Developer\Node\twitter-banner\node_modules\canvas\lib\bindings.js:3:7
      at C:\Users\pandadev\Documents\Developer\Node\twitter-banner\node_modules\canvas\lib\canvas.js:9:7
      at C:\Users\pandadev\Documents\Developer\Node\twitter-banner\node_modules\canvas\index.js:1:7

Bun v1.1.15 (Windows x64)
============================================================
Bun v1.1.15 (b23ba1fe) Windows x64
Args: "bun" "run" "src\generate.ts"
Features: jsc dotenv transpiler_cache(3) 
Builtins: "bun:main" "node:assert" "node:buffer" "node:events" "node:fs" "node:http" "node:https" "node:path" "node:stream" "node:string_decoder" "node:url" "node:util" "node:util/types" "node:zlib" 
Elapsed: 366ms | User: 0ms | Sys: 0ms
RSS: 0.14GB | Peak: 0.14GB | Commit: 0.16GB | Faults: 36396
```

### Stack Trace (bun.report)

Bun v1.1.15 ([`b23ba1f`](<https://github.com/oven-sh/bun/tree/b23ba1fe1805891cdf4566331f0c8e87d0a5b758>)) on windows x86_64 [RunCommand]

Segmentation fault at address 0x7FFC67DF0C70

- *1 unknown/js code*
- ??? at `0x5177c` in ntdll.dll
- ??? at `0x514d7` in ntdll.dll
- ??? at `0x5117d` in ntdll.dll
- ??? at `0x17fab` in KERNEL32.DLL
- ??? at `0x1bed8` in ucrtbase.dll
- ??? at `0x1c099` in ucrtbase.dll
- [`Global.zig:123`](<https://github.com/oven-sh/bun/blob/b23ba1fe1805891cdf4566331f0c8e87d0a5b758/src/Global.zig#L123>): `exitWide`
- [`Global.zig:106`](<https://github.com/oven-sh/bun/blob/b23ba1fe1805891cdf4566331f0c8e87d0a5b758/src/Global.zig#L106>): `exit`
- [`javascript.zig:1094`](<https://github.com/oven-sh/bun/blob/b23ba1fe1805891cdf4566331f0c8e87d0a5b758/src/bun.js/javascript.zig#L1094>): `eventLoop`

**Comment by 0PandaDEV** [comment]
upgrading did not fix the problem

**Comment by zoto-ff** [comment]
`canvas` depends on V8 C++ API which is not implemented now, but will be added soon.
try to use [`@napi-rs/canvas`](https://www.npmjs.com/package/@napi-rs/canvas) or smth else

**Comment by Jarred-Sumner** [comment]
Duplicate of https://github.com/oven-sh/bun/issues/5835

**Comment by Jarred-Sumner** [comment]
Please follow along in https://github.com/oven-sh/bun/issues/5835
